app/models/katello/rhsm_fact_parser.rb

```diff
@@ -21,9 +21,15 @@ def support_interfaces_parsing?
     end
 
     def get_facts_for_interface(interface)
+      # if slave then permanent_mac_address contains the physical mac
```

Comment:
kind of a nitpick on the organization (I'm aware that this was how the patch was received)

notice below we call `get_rhsm_ip(interface)` ? I'd like to see similar used here ie

```
'macaddress' => interface_mac(interface)
```

And the implementation of that method might look like:

```
facts["net.interface.#{interface}.permanent_mac_address"] || facts["net.interface.#{interface}.mac_address"]
```

Reply:
@jturel yeah you're right, this is much more elegant and idiomatic Ruby :)  Hope you don't mind I changed the method name to echo the existing `get_rhsm_ip` method.